Application/Source/Util/ActivityIndicator.swift

```diff
@@ -28,7 +28,8 @@ let successIndicator = AnyObserver<String> { event in
         hud.indicatorView = JGProgressHUDSuccessIndicatorView()
         hud.textLabel.text = value
         hud.backgroundColor = UIColor(white: 0, alpha: 70%)
-        hud.show(in: UIApplication.shared.keyWindow ?? UIWindow())
+        guard let keyWindow = UIApplication.shared.keyWindow else {return}
```

Comment:
Please put spaces around the `return` statement.

Another thing, wouldn't it be better to put the guard at the beginning, so that we make sure that we do no extra work?